torch/_dynamo/guards.py

```diff
@@ -1731,7 +1736,21 @@ def cleanup_builder(weak_b):
             ):
                 continue
 
+            # Delay the duplicate input guards
+            if not config.enable_cpp_guard_manager:
+                if isinstance(guard.create_fn, functools.partial):
+                    fn = guard.create_fn.func
+                    if fn is GuardBuilder.DUPLICATE_INPUT:
+                        duplicate_input_guards.add(guard)
+                        continue
+
             guard.create(builder)
+
+        # Build duplicate input guards
+        if not config.enable_cpp_guard_manager:
+            for guard in duplicate_input_guards:
```

Comment:
Sets have non-determinstic order when you iterate over them.  Need a non-set, or `sorted`.